pkg/store/trtl/trtl.go

```diff
@@ -720,6 +720,111 @@ func (s *Store) DeleteAnnouncementMonth(ctx context.Context, date string) (err e
 	return nil
 }
 
+//===========================================================================
+// ActivityStore Implementation
+//===========================================================================
+
+// RetrieveActivityMonth returns the activity month record for the given date
+// timestamp in the format YYYY-MM.
+func (s *Store) RetrieveActivityMonth(ctx context.Context, date string) (m *bff.ActivityMonth, err error) {
+	if date == "" {
+		return nil, storeerrors.ErrEntityNotFound
+	}
+
+	// Get the key by creating an intermediate activity month to ensure that
+	// validation and key creation always happens the same way.
+	m = &bff.ActivityMonth{Date: date}
+	key, err := m.Key()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := utils.WithDeadline(ctx)
+	defer cancel()
+	request := &pb.GetRequest{
+		Key:       key,
+		Namespace: wire.NamespaceActivities,
+	}
+	var reply *pb.GetReply
+	if reply, err = s.client.Get(ctx, request); err != nil {
+		if status.Code(err) == codes.NotFound {
+			return nil, storeerrors.ErrEntityNotFound
+		}
+		return nil, err
+	}
+
+	if err = proto.Unmarshal(reply.Value, m); err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+// UpdateActivityMonth creates a new activity month record if it doesn't already
+// exist or replaces the existing record.
+func (s *Store) UpdateActivityMonth(ctx context.Context, m *bff.ActivityMonth) (err error) {
```

Comment:
Ok - so just so we're on the same page; there is a race condition in Retrieve than Update patterns; e.g. if go routine A retrieves, then routine B retrieves, then B updates, then A updates; B's updates will be lost. This shouldn't be a problem given there is only a single go routine updating the store from the BFF (e.g. the activity subscriber); but I wanted to mention it.

Reply:
Definitely, thanks for pointing that out.